gbpservice/nfp/service_vendor_agents/vyos/oc_config_server/stats_parser.py

```diff
@@ -0,0 +1,253 @@
+                    return interface
+
+            except KeyError as keyerr:
+                logger.error('Unable to Parse Stats Data, ' +
+                    'KeyError: {}'.format(keyerr))
+        return None
+
+    def parse_firewall_stats(self, interface, raw_stats):
+        firewall  = {}
+        firewalls = []
+        firewall_start = False
+        table = False
+        status = None
+        rule_keys = ['rulepriority', 'packets', 'bytes', 'action',
+                    'source', 'destination']
+
+        try:
+            for line in raw_stats.split('\n'):
+                words = line.split()
+                if 'IPv4 Firewall' in line:
+                    firewall_start = True
+                if 'Active on' in line and interface in line and firewall_start:
+                    status = "Active"
+                    (interface, direction) = words[2][1:-1].split(',')
+                    firewall['interface'] = interface
+                    firewall['dir'] = direction
+                    firewall['rules'] = []
+                elif len(words) >= 4:
+                    if words[3] in ['ACCEPT', 'DROP'] and status == "Active":
+                        table = True
+                        rule = dict(zip(rule_keys, words))
+                        firewall['rules'].append(rule)
+                elif table and status == "Active":
+                    command = ('/opt/vyatta/bin/vyatta-show-firewall.pl "all_all" ' +
+                               '/opt/vyatta/share/xsl/show_firewall_detail.xsl')
+                    show_fw_data = self.run_command(command)
+                    firewall = self.add_protocol_and_dest_port_info(firewall, show_fw_data)
+                    logger.info("packed firewall \n %s" % firewall)
+                    firewalls.append(firewall)
+                    break;
+
+        except KeyError as keyerr:
+            logger.error('Unable to Parse Firewall Stats Data, ' +
+                    'KeyError: {}'.format(keyerr))
+
+        except IndexError as inderr:
+            logger.error('Unable to Parse Firewall Stats Data, ' +
+                    'IndexError: {}'.format(inderr))
+
+        return firewalls
+
+    def add_protocol_and_dest_port_info(self, firewall, show_fw_data):
+        firewall_started = False
+        firewall_info_started = False
+        firewall_matcher = "Active on (" + firewall['interface']
+        firewall_info_end = "-------------"
+        firewall_info = []
+        for line in show_fw_data.split('\n'):
+            if "IPv4 Firewall" in line:
+                firewall_started = True
+            if firewall_matcher in line:
+                firewall_info_started = True
+            if firewall_started and firewall_info_started:
+                firewall_info.append(line)
+            if firewall_started and firewall_info_started and firewall_info_end in line:
+                break;
+        try:
+            for rule in firewall.get('rules', None):
```

Comment:
This will throw error "Nonetype object is not iterable"
when rules are not present.
Use empty list instead of None